### PR TITLE
feat: group-drop-down-component-changes

### PR DESCRIPTION
**Pull request by PritishBudhiraja**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

image

image

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Review by Riddhiagrawal001 — DISMISSED** [review]

File: src/components/InputFields.resi [hunk 1]
```diff
@@ -43,16 +43,16 @@ let selectInput: (
   ~fixedDropDownDirection: SelectBox.direction=?,
   ~customButton: React.element=?,
   ~buttonType: Button.buttonType=?,
-  ~dropdownCustomWidth: string=?,
+  ~dropdownCustomWidth: Js.String2.t=?,
```
Comment (JeevaRamu0104) [inline src/components/InputFields.resi:46]: why `string`  type changed to `Js.String2.t`

File: src/components/InputFields.resi [hunk 2]
```diff
@@ -43,16 +43,16 @@ let selectInput: (
...
   ~allowButtonTextMinWidth: bool=?,
-  ~setExtSearchString: (string => string) => unit=?,
-  ~textStyleClass: string=?,
+  ~setExtSearchString: ('b => string) => unit=?,
```
Comment (JeevaRamu0104) [inline src/components/InputFields.resi:48]: can we avoid using generic type

File: src/components/SelectBox.res [hunk 3]
```diff
@@ -1104,6 +1107,123 @@ module BaseSelectButton = {
...
+          textColorClass={option.textColor}
+          customMarginStyle=customMarginStyleOfListItem
+        />
+
+      if !descriptionOnHover {
+        switch option.description {
+        | Some(str) =>
+          <div key={i->string_of_int} className="flex flex-row">
+            listItemComponent
+            {if isHorizontal {
```
Comment (JeevaRamu0104) [inline src/components/SelectBox.res:1178]: can we `UiUtils.RenderIf`

File: src/components/InputFields.resi (hunk 2 above)
Comment (PritishBudhiraja) [reply src/components/InputFields.resi:48]: These files are generated by rescript extension.

File: src/components/InputFields.resi (hunk 1 above)
Comment (PritishBudhiraja) [reply src/components/InputFields.resi:46]: These files are generated by rescript extension.

File: src/components/SelectBox.res (hunk 3 above)
Comment (PritishBudhiraja) [reply src/components/SelectBox.res]: Addressed

**Review by JeevaRamu0104 — APPROVED** [review]

**Review by Riddhiagrawal001 — APPROVED** [review]

**Pull request merged**